sim/battle.ts

```diff
@@ -19,6 +19,11 @@ interface FaintedPokemon {
 	effect: Effect | null;
 }
 
+interface MoveHitData {
+	crit: {[key: string]: boolean};
```

Comment:
`key` is really vague, and appears to be `": "` plus the Pokémon nickname, which is probably unintentional.

Reply:
Oh yeah it should be `.slice(0, 3`

Reply:
What would you call “p1a”, “p2b”, etc.? “targetingid”?

Reply:
`slotid`?

Reply:
Also I wouldn't do this at all. :[